core/domain/job_services.py

```diff
@@ -96,19 +107,25 @@ def train_classifier(algorithm_id, training_data):
     """
     classifier = algorithm_registry.Registry.get_classifier_by_algorithm_id(
         algorithm_id)
+    if classifier.version != algorithm_version:
+        logging.warning(
+            'Classifier version %d mismatches algorithm version %d received '
+            'in job data', classifier.version, algorithm_version)
+        return None
```

Comment:
Is there a test to verify that this condition is properly hit as expected?

Reply:
Actually, there was no test file for `job_services` module. I just added one with a bunch of test cases for every function and every possible scenarion

Reply:
Thanks!